### PR TITLE
Info dump 55/66: Dungeon music

### PR DESCRIPTION
**Pull request by End45**
Adds some functions related to dungeon music and playing audio. The bulk of my notes about the audio system is yet to be dumped though (section 61: Sky Jukebox & Music).

I also updated the functions that refer to debug flags to be more accurate.

### CODE REVIEW

File: headers/types/common/enums.h [hunk 1]
```diff
@@ -3722,4 +3722,43 @@ enum wan_source_type {
...
+    DEBUG_FLAG_SOUND_OFF = 2,       // Disables all sounds?
+    DEBUG_FLAG_BGM_OFF = 3,         // Background music won't be played
+    DEBUG_FLAG_SE_OFF = 4,          // Sound effects won't be played
+    DEBUG_FLAG_STAGE_NPC_DUMMY = 5, // Unknown purpose
+    // Was supposed to throw an error if text overflowed the textbox it was contained in, but it
+    // does nothing in the final game.
+    DEBUG_FLAG_TEXT_LIMIT_CHECK = 6,
+    DEBUG_FLAG_NO_CHEAT_CHECK = 7,             // Unknown purpose
+    DEBUG_FLAG_NO_PLUNGE_CHECK = 8,            // Unknown purpose
+    DEBUG_FLAG_DUNGEON_INFINITE_COMEBACK = 10, // Unknown purpose
```
Comment (UsernameFodder) [inline headers/types/common/enums.h:3738]: This sounds like infinite revives within a dungeon? Idk how you verified the purpose of each debug flag though.

File: symbols/overlay29.yml [hunk 2]
```diff
@@ -1003,6 +1003,17 @@ overlay29:
         Sets the dungeon PRNG to use the primary LCG for subsequent random number generation.
         
         No params.
+    - name: DungeonMusicIdToMusicId
```
Comment (UsernameFodder) [inline symbols/overlay29.yml:1006]: "Dungeon music ID" is kind of a confusing name IMO. Can we use the terminology of the `floor_properties` field? Something like `MusicTableIdxToMusicId`?

File: headers/functions/arm9.h [hunk 3]
```diff
@@ -90,13 +90,13 @@ void FileSeek(struct file_stream* file, int offset, int whence);
...
-uint32_t GetDebugFlag1(uint32_t flag_id);
-void SetDebugFlag1(uint32_t flag_id, uint32_t val);
+bool GetDebugFlag(enum debug_flag flag);
+void SetDebugFlag(enum debug_flag flag, bool val);
 int AppendProgPos(char* str, struct prog_pos_info* prog_pos, const char* msg);
 void DebugPrintTrace(const char* msg, struct prog_pos_info* prog_pos);
 void DebugPrint0(const char* fmt, ...);
-uint32_t GetDebugFlag2(uint32_t flag_id);
-void SetDebugFlag2(uint32_t flag_id, uint32_t val);
+bool GetDebugLogFlag(enum debug_flag flag);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:98]: Should this be:

```suggestion
bool GetDebugLogFlag(enum debug_log_flag flag);
```

?

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/types/common/enums.h (hunk 1 above)
Comment (End45) [reply headers/types/common/enums.h:3738]: Nope, I already tested it and it does not prevent you from fainting.

File: symbols/overlay29.yml (hunk 2 above)
Comment (End45) [reply symbols/overlay29.yml:1006]: Sure

File: headers/functions/arm9.h (hunk 3 above)
Comment (End45) [reply headers/functions/arm9.h:98]: Fixed, thanks.

**Pull request merged**